test/integration/orm_action_execution.go

```diff
@@ -0,0 +1,216 @@
+			kubeClient = f.GetKubeClient("orm-action-executor")
+
+			var err error
+			dynamicClient, err = dynamic.NewForConfig(kubeConfig)
+			if err != nil {
+				framework.Failf("Failed to generate dynamic client for kubernetes test cluster: %v", err)
+			}
+
+			s := app.NewVMTServer()
+			kubeletClient := s.CreateKubeletClientOrDie(kubeConfig, kubeClient, "", "busybox", map[string]set.Set{}, true)
+			apiExtClient, err := apiextclient.NewForConfig(kubeConfig)
+			if err != nil {
+				glog.Fatalf("Failed to generate apiExtensions client for kubernetes target: %v", err)
+			}
+			ormClient := resourcemapping.NewORMClient(dynamicClient, apiExtClient)
+
+			probeConfig := createProbeConfigOrDie(kubeClient, kubeletClient, dynamicClient)
+
+			discoveryClientConfig := discovery.NewDiscoveryConfig(probeConfig, nil, app.DefaultValidationWorkers,
+				app.DefaultValidationTimeout, aggregation.DefaultContainerUtilizationDataAggStrategy,
+				aggregation.DefaultContainerUsageDataAggStrategy, ormClient, app.DefaultDiscoveryWorkers, app.DefaultDiscoveryTimeoutSec,
+				app.DefaultDiscoverySamples, app.DefaultDiscoverySampleIntervalSec)
+			actionHandlerConfig := action.NewActionHandlerConfig("", nil, nil,
+				cluster.NewClusterScraper(kubeClient, dynamicClient, false, nil, ""),
+				[]string{"*"}, ormClient, false, true, 60, gitops.GitConfig{})
+			actionHandler = action.NewActionHandler(actionHandlerConfig)
+
+			// Kubernetes Probe Discovery Client
+			discoveryClient := discovery.NewK8sDiscoveryClient(discoveryClientConfig)
+			_, _, err = discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
+			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+		}
+
+	})
+
+	Describe("ORM test for namespace scope", func() {
+		It("should match with the new CR", func() {
+			// 1. Get the deployment
+			dep, err := kubeClient.AppsV1().Deployments(ormtest_namespace).Get(context.TODO(), namespacescope_operand_deployment_name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Error gettting the deployment for ORM test")
+
+			// 2. Get the CR and backup the resource value
+			operatorCR, err := dynamicClient.Resource(groupVersionRes4nsscope).Namespace(ormtest_namespace).Get(context.TODO(), namespacescope_operator_cr_name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Error gettting the operator CR")
+			origCRValue4NsScope, _, err = util.NestedField(operatorCR, "destPath", namespacescope_operand_path_in_cr)
+
+			// 3. Build and execute resize action on the deployment
+			targetSE := newResizeWorkloadControllerTargetSE(dep)
+			resizeupAction, desiredPodSpec := newResizeActionExecutionDTO(targetSE, ORM_RESIZE_UP, &dep.Spec.Template.Spec)
+			_, err = actionHandler.ExecuteAction(resizeupAction, nil, &mockProgressTrack{})
+			if err != nil {
+				framework.Failf("Failed to execute resizing action for ORM namespace case")
+			}
+
+			// 4. Set the revert flag for afterEach
+			needRevertCR4NsScope = true
```

Comment:
I recommend to let the revert logic run irrespective of action being successful or not. You don't know at what stage the action failed, it might happen that the resource was updated but action failed for a secondary reason.
There is no harm in updating the resource with its original values, even if they are still the same.

Reply:
sure,  will remove this flag